cli/flox-rust-sdk/src/models/lockfile.rs

```diff
@@ -350,24 +352,27 @@ impl LockedManifestCatalog {
 
     /// Creates package groups from a flat map of install descriptors
     ///
-    /// A group is created for each unique combination of (descriptor.package_group ｘ descriptor.system).
+    /// A group is created for each unique combination of (descriptor.package_group ｘ descriptor.systems).
+    /// If descriptor.systems is None, a group with default_system is created for each package_group.
     /// Each group contains a list of package descriptors that belong to that group.
     ///
     /// `seed_lockfile` is used to provide existing derivations for packages that are already locked,
     /// e.g. by a previous lockfile.
     /// These packages are used to constrain the resolution.
     /// If a package in `manifest` does not have a corresponding package in `seed_lockfile`,
     /// that package will be unconstrained, allowing a first install.
-    fn collect_package_groups<'manifest>(
-        manifest: &'manifest TypedManifestCatalog,
+    fn collect_package_groups(
+        manifest: &TypedManifestCatalog,
         seed_lockfile: Option<&LockedManifestCatalog>,
-    ) -> impl Iterator<Item = PackageGroup> + 'manifest {
+        default_system: &str,
```

Comment:
suggestion(nb): accept a list of systems to allow the _resolution_ default to be "all systems".
For pkgdb we decided for a single system due to scrape times, using the catalog locking more than one system by default shouldn't change the performance profile.

It's a reasonably small change to do right here but not exactly required either at this point

Reply:
That's a good point might as well do now!